### ISSUE TITLE
Allocation error handling

### ISSUE THREAD

**Issue by elfring** [emersion/mrsh#80]
Would you like to add more error handling for return values from functions like the following?
- [malloc](http://pubs.opengroup.org/onlinepubs/9699919799/functions/malloc.html "Memory allocation") ⇒ [interactive_next](https://github.com/emersion/mrsh/blob/eaf18a7e54797ec42929f370acafb8ab2a549e4e/frontend/readline.c#L30)
- [strdup](http://pubs.opengroup.org/onlinepubs/9699919799/functions/strdup.html "Duplicate a string.") ⇒ [mrsh_hashtable_set](https://github.com/emersion/mrsh/blob/442073258468d51704f72b33ba32d4b7be17227f/hashtable.c#L30)

**Comment by emersion** [comment]
Yeah, this is a TODO. I'd merge patches to do so. Adding this to the core mrsh library has higher priority than adding it to executables.

**Comment by elfring** [comment]
How do you think about to improve [static source code analysis](https://en.wikipedia.org/wiki/List_of_tools_for_static_code_analysis#C.2C_C.2B.2B "Tools for code analysis") also for this software?

**Comment by emersion** [comment]
What do you mean?

I run Clang's static analyzer from time to time. I'd prefer not to run it for each pull request because it's slow and has false positives. Do you have better ideas?

Related: I also run sanitizers as part of CI: Arch builds run tests with ASan and UBSan enabled.

**Comment by elfring** [comment]
* Are you interested to apply [aspect-oriented software development](http://aspectc.org/ "Software development with AspectC++?")?
* How do you think about to encapsulate error detection and corresponding exception handling as [a reusable aspect](https://en.wikipedia.org/wiki/Aspect_%28computer_programming%29 "Combination of a pointcut with advice") in your software?

**Comment by emersion** [comment]
No, sorry, I'd prefer to stick with the current style. Readability and KISS first.

**Comment by ddevault** [comment]
Could run static analysis on each commit rather than each PR. Would require some builds.sr.ht changes.
